core/src/test/java/org/apache/iceberg/TestDeleteFiles.java

```diff
@@ -68,38 +68,46 @@ public class TestDeleteFiles extends TableTestBase {
                   ))
           .build();
 
-  @Parameterized.Parameters(name = "formatVersion = {0}")
+  private final String branch;
+
+  @Parameterized.Parameters(name = "formatVersion = {0}, branch = {1}")
   public static Object[] parameters() {
-    return new Object[] {1, 2};
+    return new Object[][] {
+      new Object[] {1, "main"},
+      new Object[] {1, "testBranch"},
+      new Object[] {2, "main"},
+      new Object[] {2, "testBranch"}
+    };
   }
 
-  public TestDeleteFiles(int formatVersion) {
+  public TestDeleteFiles(int formatVersion, String branch) {
     super(formatVersion);
+    this.branch = branch;
   }
 
   @Test
   public void testMultipleDeletes() {
-    table.newAppend().appendFile(FILE_A).appendFile(FILE_B).appendFile(FILE_C).commit();
+    Snapshot append =
```

Comment:
Style (and minor): I'd prefer to prepare the change and then pass it in to avoid this formatting. I think it would be cleaner and more readable like this:

```java
SnapshotProducer action = table.newAppend().appendFile(FILE_A).appendFile(FILE_B).appendFile(FILE_C);
commit(table, action, branch);

Snapshot append = latestSnapshot(readMetadata(), branch);
```

I also prefer to call `latestSnapshot` separately to get the `Snapshot`. We don't need to put too much on `commit`.

Reply:
Thinking about this a bit more, returning the committed `Snapshot` from `commit` is fine. Up to you what to do here, I'm probably being pedantic.

Reply:
Sure, I left it as is for now.